### ISSUE TITLE
NPC Sparring - Instant Respawns

### ISSUE THREAD

**Issue by stormhalter** [jkachhad/Stormhalter#1201]
Instead of multi-box sparring and having player deaths report to the Death bot, the new sparring high ground seems to be utilizing instant respawn NPCs to gain the same buff benefits without the same level of reporting. [Reported by Vernak]

---
[View on Discord](https://discord.com/channels/1056432310632841276/1530613115635499050/1530613115635499050)

---
[View on Discord](https://discord.com/channels/577503035526348810/1532151247627489480/1532151247627489480)

**Comment by Tawnos-LOK** [comment]
Related, people are sparring invulnerable vendors as well

**Comment by jkachhad** [comment]
I think the best fix for this would be make it so that any `CreatureEntity` that awards 0 experience, also does not award any skill when attacked.   We can then configure all vendors to get 0 exp, but still allow them to be killed for respawning.
I can go through it add it for all damage and direct spell damage.

The other thought it to implement it per subregion, Town does not award any skill, unless inside fighting ring?

**Comment by Tawnos-LOK** [comment]
> I think the best fix for this would be make it so that any `CreatureEntity` that awards 0 experience, also does not award any skill when attacked.   We can then configure all vendors to get 0 exp, but still allow them to be killed for respawning.
> I can go through it add it for all damage and direct spell damage.
> 
> The other thought it to implement it per subregion, Town does not award any skill, unless inside fighting ring?

I agree with the 0 XP = 0 skill part.  Vendors shouldn't give melee or magic skill.  On the last part, it would be ok if there was no town skilling for melee outside a sparring ring but for magic, I could see if people wanted to gain magic skill for their self buffs, recall ring summons, etc. In town, albeit at a slower rate than kill buff.  Debatable.
